datafusion/optimizer/src/analyzer/mod.rs

```diff
@@ -17,10 +17,12 @@
 
 mod count_wildcard_rule;
 mod inline_table_scan;
+pub(crate) mod type_coercion;
 
 use crate::analyzer::count_wildcard_rule::CountWildcardRule;
 use crate::analyzer::inline_table_scan::InlineTableScan;
 
+use crate::analyzer::type_coercion::TypeCoercion;
```

Comment:
🎉

Reply:
👍

Reply:
we can also close this issue https://github.com/apache/arrow-datafusion/issues/3582
cc @alamb @jackwener

Reply:
FYI @mingmwang  I think you have discussed this as well in the past. It is great to see it has finally been done -- thanks @jackwener !